apache/superset
# chore(Dashboard): Implement dedicated message for missing dataset

### SUMMARY
Implements a dedicated "Missing dataset" message to replace the generic "Unexpected error".

### BEFORE
<img width="674" alt="Screen Shot 2021-10-01 at 10 48 24 AM" src="https://user-images.githubusercontent.com/81597121/135667085-810e2816-9793-4975-9765-9a04c96a5fb4.png">

### AFTER
![Screen Shot 2021-10-04 at 19 12 21](https://user-images.githubusercontent.com/60598000/135886536-410b09a0-680d-43e3-8680-12ff4ee81d53.png)


### TESTING INSTRUCTIONS
1. Open a Dashboard with a dataset that does not exist
2. Observe the error message

### ADDITIONAL INFORMATION


- [x] Has associated issue: Fixes #16938
